src/elements/components/download-button.tsx

```diff
@@ -36,43 +42,147 @@ const getHostFromUrl = (url: string) => {
         }
         return parsedUrl.hostname;
     } catch (e) {
-        console.error(e);
+        console.debug(e);
         return 'an unknown hoster';
     }
 };
 
-export const DownloadButton = ({ url, resource }: DownloadButtonProps) => {
-    const isExternal = !url.startsWith(
-        'https://objectstorage.us-phoenix-1.oraclecloud.com/n/ax6ygfvpvzka/b/open-modeldb-files/'
-    );
-    const host = getHostFromUrl(url);
+const getHostIcon = (host: string) => {
+    switch (host) {
+        case 'GitHub':
+            return <SiGithub className="block" />;
+        case 'Google Drive':
+            return <SiGoogledrive className="block" />;
+        case 'OneDrive':
+            return <SiMicrosoftonedrive className="block" />;
+        case 'Mega':
+            return <SiMega className="block" />;
+        case 'Dropbox':
+            return <SiDropbox className="block" />;
+        default:
+            return <FiExternalLink className="block" />;
+    }
+};
+
+const getIsExternal = (url: string) => {
+    return !url.startsWith('https://objectstorage.us-phoenix-1.oraclecloud.com/n/ax6ygfvpvzka/b/open-modeldb-files/');
+};
+
+export const DownloadButton = ({ resource, readonly, onChange }: DownloadButtonProps) => {
+    const [selectedMirror, setSelectedMirror] = useState<string | undefined>(resource.urls[0]);
```

Comment:
Can a resource have 0 URLs? I would say no. You probably did this because you ended up with those resources while removing URLs, right? I think instead of allowing resources with 0 URLs, we should just remove the resource if there are no URLs for it. Or we don't allow removing URLs if there's only one.